test/NuGet.Core.Tests/NuGet.Packaging.Test/DependenciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,446 @@
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            //Arrange
+            var fileStrings = new string[]
+            {
+                "ref/net20/test.dll",
+                "ref/net35/test.dll",
+                "ref/net40/test.dll",
+                "ref/net45/test.dll",
+                "ref/netstandard1.0/test.dll",
+                "ref/netstandard1.3/test.dll",
+                "ref/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithoutDependenciesForEachTFMInLib_ShouldThrowOneWarningCode()
+        {
+            //Arrange
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            var fileStrings = new string[]
+            {
+                "lib/net20/test.dll",
+                "lib/net35/test.dll",
+                "lib/net40/test.dll"
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            var testStringExact = AnalysisResources.DependenciesGroupsForEachTFMHasNoExactMatch + "\n" +
+                "- Add lib or ref assemblies for the net45 target framework\r\n" +
+                "- Add lib or ref assemblies for the netstandard1.0 target framework\r\n" +
+                "- Add lib or ref assemblies for the netstandard1.3 target framework\r\n" +
+                "- Add lib or ref assemblies for the netstandard2.0 target framework\r\n";
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5128 && p.Message == testStringExact));
```

Comment:
Just FYI if you don't have time to change, it's better to do something like:
```csharp
var warning = issues.Single(p => p.Code == NuGetLogCode.NU5128);
Assert.Equal(testStringExact, warning.Message);
```

This way

1. If `GenerateWarnings` generates multiple `NU5128` warnings, it fails
2. If the string is wrong, the Assert class can tell us what the actual and expected strings are, and where in the string the difference was.